### DISCUSSION TITLE
how do i make onefile from .spec?

### DISCUSSION THREAD

**Discussion by wookidooki8** [pyinstaller/pyinstaller#7351] [Help] [answered]
https://stackoverflow.com/questions/67868395/hiding-python-source-code-to-prevent-people-from-seeing-it-the-whole-and-prope

im following the exact same on the above link.
i got a few .c and .pyd files to put it all together and make it one file but it gives me this error

    option(s) not allowed:
      --onedir/--onefile
    makespec options not valid when a .spec file is given

i found [this post ](https://github.com/pyinstaller/pyinstaller/issues/7203) and figured its no longer supported.
so is there any way to do what i want? (putting *.c and *.pyd all togther in one file)

**Comment by bwoodsend** [comment, marked as answer]
You need to edit the spec file to convert it from a onedir spec file to a onefile one. If you run `pyi-makespec a.py` and `pyi-makespec --onefile b.py` then compare the generated `a.spec` and `b.spec`, you should be able to figure out what changes you need to make to your own spec file.

> i found https://github.com/pyinstaller/pyinstaller/issues/7203and figured its no longer supported.

It was never supported - it just used to ignore the flag rather than raise an error. I see the post on stackoverflow incorrectly says that you can - it's wrong, I'll comment there.

**Reply by wookidooki8** [reply]
thank you for the answer! thumbs up!!
